src/site/markdown/cache-configuration.md

```diff
@@ -0,0 +1,188 @@
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+# Maven Cache Configuration Enhancement
+
+This document describes the enhanced cache configuration functionality in Maven's DefaultRequestCache.
+
+## Overview
+
+The DefaultRequestCache has been enhanced to support configurable reference types and cache scopes through user-defined selectors. This allows fine-grained control over caching behavior for different request types.
+
+## Key Features
+
+### 1. Early Return for ProtoSession
+- The `doCache` method now returns early when the request session is not a `Session` instance (e.g., `ProtoSession`)
+- This prevents caching attempts for non-session contexts
+
+### 2. Configurable Cache Behavior
+- Cache scope and reference type can be configured via session user properties
+- Configuration uses CSS-like selectors to match request types
+- Supports parent-child request relationships
+
+### 3. Backward Compatibility
+- Existing `CacheMetadata` interface still supported
```

Comment:
in general, write in present tense. The reader will want to know how this operates. That is, "this is what the cache does" rather than "this is how the cache has changed".